pkg/statistics/handle/storage/stats_read_writer.go

```diff
@@ -230,24 +230,19 @@ func (s *statsReadWriter) SaveColOrIdxStatsToStorage(
 	return
 }
 
-// SaveMetaToStorage saves stats meta to the storage.
+// SaveMetasToStorage saves stats meta to the storage.
 // Use the param `refreshLastHistVer` to indicate whether we need to update the last_histograms_versions in stats_meta table.
-func (s *statsReadWriter) SaveMetaToStorage(
-	tableID, count, modifyCount int64,
-	source string,
+func (s *statsReadWriter) SaveMetasToStorage(
 	refreshLastHistVer bool,
+	metaUpdates ...statstypes.MetaUpdate,
 ) (err error) {
-	var statsVer uint64
+	if len(metaUpdates) == 0 {
+		return nil
+	}
 	err = util.CallWithSCtx(s.statsHandler.SPool(), func(sctx sessionctx.Context) error {
-		statsVer, err = SaveMetaToStorage(sctx, tableID, count, modifyCount, refreshLastHistVer)
-		if err != nil {
-			return err
-		}
-		return err
+		return SaveMetasToStorage(sctx, refreshLastHistVer, metaUpdates)
 	}, util.FlagWrapTxn)
-	if err == nil && statsVer != 0 {
-		s.statsHandler.RecordHistoricalStatsMeta(statsVer, source, false, tableID)
-	}
+	// NOTE: recording historical stats meta is deprecated.
```

Comment:
It’s almost deprecated, but not entirely. We’ve just turned off this feature by default. So I think it’s better to keep this code, as it already supports (fake🤣) batch mode.